conans/client/tools/scm.py

```diff
@@ -146,4 +159,144 @@ def get_branch(self):
+        runner = runner or runner_no_strip
+        super(SVN, self).__init__(folder=folder, runner=runner, *args, **kwargs)
+
+    @staticmethod
+    def get_version():
+        try:
+            out, err = subprocess.Popen(["svn", "--version"], stdout=subprocess.PIPE).communicate()
+            version_line = decode_text(out).split('\n', 1)[0]
+            version_str = version_line.split(' ', 3)[2]
+            return Version(version_str)
+        except Exception as e:
+            raise ConanException("Error retrieving SVN version: '{}'".format(e))
+
+    @property
+    def version(self):
+        if not hasattr(self, '_version'):
+            version = SVN.get_version()
+            setattr(self, '_version', version)
+        return getattr(self, '_version')
+
+    def run(self, command):
+        # Ensure we always pass some params
+        extra_options = " --no-auth-cache --non-interactive"
+        if not self._verify_ssl:
+            if self.version >= SVN.API_CHANGE_VERSION:
+                extra_options += " --trust-server-cert-failures=unknown-ca"
+            else:
+                extra_options += " --trust-server-cert"
+        return super(SVN, self).run(command="{} {}".format(command, extra_options))
+
+    def checkout(self, url, revision="HEAD"):
+        output = ""
+        try:
+            self._check_svn_repo()
+        except ConanException:
+            output += self.run('co "{url}" .'.format(url=url))
+        else:
+            assert url.lower() == self.get_remote_url().lower(), \
+                "%s != %s" % (url, self.get_remote_url())
+            output += self.run("revert . --recursive")
+        finally:
+            output += self.update(revision=revision)
+        return output
+
+    def update(self, revision='HEAD'):
+        self._check_svn_repo()
+        return self.run("update -r {rev}".format(rev=revision))
+
+    def excluded_files(self):
+        self._check_svn_repo()
+        excluded_list = []
+        output = self.run("status --no-ignore")
+        for it in output.splitlines():
+            if it[0] == 'I':  # Only ignored files
+                filepath = it[8:].strip()
+                excluded_list.append(os.path.normpath(filepath))
+        return excluded_list
+
+    def get_remote_url(self):
+        return self.run("info --show-item url").strip()
+
+    def get_qualified_remote_url(self):
+        # Return url with peg revision
+        url = self.get_remote_url()
+        revision = self.get_last_changed_revision()
+        return "{url}@{revision}".format(url=url, revision=revision)
+        
+    def is_local_repository(self):
+        url = self.get_remote_url()
+        return url.startswith(self.file_protocol) and \
+               os.path.exists(unquote(url[len(self.file_protocol):]))
+
+    def is_pristine(self):
+        # Check if working copy is pristine/consistent
+        output = self.run("status -u -r {} --xml".format(self.get_revision()))
```

Comment:
Please don't check against remote repository. We only have to check the local repository for local changes & so on. The remote repository will change after some time against the revision we put into our working copy.

Use `svn status -v -r {} --xml` to get the full list of changes in the repository. Keep the search for `wc-status` and  remove `repos-status`. Also add `item.get('revision')` and find min and max value. When they are the same, there is no mixed revision.

Reply:
You are right, I'm mixing two different things:

1. The repo itself, it will be _pristine_ given your considerations.
2. The SCM feature that makes use of this function. The feature uses this SVN class to handle a working copy, in this use case we do want to check against remote, because SCM wants the working copy to be in a given revision and it won't be ok if it is _pristine_ (previous point) but not in the instructed revision.

Does it make sense to you?

Reply:
> 2. The SCM feature that makes use of this function. The feature uses this SVN class to handle a working copy, in this use case we do want to check against remote, because SCM wants the working copy to be in a given revision and it won't be ok if it is _pristine_ (previous point) but not in the instructed revision.

Why do you want to contact the remote? When I check, that the working copy is a svn revision without any change, then everything is fine. The remote would tell me, that there is a newer version, that would have some changes. But this is not important, because we just want to make sure, that the recoverability for the specific revision is given.

Reply:
To check for externals, we would need to make the following:

* Run `svn status -v --ignore-externals --xml`
  * Check `wc-status` `item` & `props` for changes...
  * Check `wc-status revision` that all the same and equal to `svn revision`
* Search for externals `svn propget svn:externals -R` & loop over all externals
  * Check, that externals is fixed to peg or operative revision
  * Run `svn status -v --ignore-externals --xml`
    * Check `wc-status` `item` & `props` for changes...
    * Check `wc-status revision` that all the same and equal to external revision

This would check for mixed revisions, inside main project and external libraries. Also all `svn:externals` must be fixed, because otherwise would a later checkout deliver a different checkout.

Reply:
Why do you specify the revision number? During `checkout` the working copy is already set to the specific revision. Just check that the `wc-status revision` is identical to your revision.

Reply:
If I do a ``commit`` but not an ``update`` I get mixed revisions in my working copy (check ``SVNToolsTestPristine::test_mixed_revisions``). I can use param ``-v`` and check that every file in my repo is in the same revision as the one retrieved with ``SVN::get_revision`` as you suggested, but it could become a huge XML output to parse for big repos. Using the ``-r`` I get only the differences.

Reply:
@climblinne have a look at the tests I've just added in the last commit. I think they cover most of the use cases.

Reply:
I think the question is, do we always have always a connection to the subversion server when we use `is_pristine` function? When I work at home, I have not always a connection to the svn server. But still it is nice to test packages and generate the version numbers.

Reply:
It would be nice to have also the following tests:
* An `svn:externals` definition without fixed revision --> should lead to error, because not fixed
* An `svn:externals`definition with pegged revision
* An `svn:externals`definition with pegged revision & operative revision --> could be different revsisions, we must use operative revision

Reply:
Also a test, where the working copy is not able to reach the repository (Just delete the local repository) would be nice. Then it would be nice, if we where still able to run `is_pristine` function.

Reply:
Also it is possible to make a single file link instead of a directory. I never used this, but may be we should test it?

Reply:
Yes, we need to write all the tests you mentioned, although some of them may be hard to address. I have created the TODO list in the PR with all of them 👍